src/user_lists/serializers.py

```diff
@@ -1,20 +1,157 @@
+from django.core.exceptions import ObjectDoesNotExist
 from rest_framework import serializers
-from utils.serializers import DefaultAuthenticatedSerializer
 
-from .models import List
+from feed.models import FeedEntry
+from feed.serializers import FeedEntrySerializer, serialize_feed_metrics
+from paper.models import Paper
+from researchhub_document.related_models.constants.document_type import PAPER
+from researchhub_document.related_models.researchhub_post_model import ResearchhubPost
 
+from .models import List, ListItem
+
+
+ 
+READ_ONLY_FIELDS = ["id", "created_date", "updated_date", "created_by", "updated_by"]
+
+LIST_WRITABLE_FIELDS = ["is_public", "name"]
+LIST_ITEM_WRITABLE_FIELDS = ["is_public", "parent_list", "unified_document"]
+
+LIST_BASE_FIELDS = READ_ONLY_FIELDS + LIST_WRITABLE_FIELDS
+LIST_ITEM_FIELDS = READ_ONLY_FIELDS + LIST_ITEM_WRITABLE_FIELDS
+
+
+class ListSerializer(serializers.ModelSerializer):
+    items_count = serializers.SerializerMethodField()
+
+    class Meta:
+        model = List
+        fields = LIST_BASE_FIELDS + ["items_count"]
+        read_only_fields = READ_ONLY_FIELDS
+
+    def get_items_count(self, obj):
+        return getattr(obj, "items_count", 0)
+
+
+class ListItemReadSerializer(serializers.ModelSerializer):
+    unified_document = serializers.SerializerMethodField(method_name="get_feed_entry")
+
+    class Meta:
+        model = ListItem
+        fields = LIST_ITEM_FIELDS
+        read_only_fields = LIST_ITEM_FIELDS
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._metrics_cache = {}
+
+    def get_feed_entry(self, obj):
+        entry = self._get_or_create_feed_entry(obj)
+        return FeedEntrySerializer(entry, context=self.context).data if entry else None
+
+    def _get_or_create_feed_entry(self, obj):
+        cached = getattr(obj.unified_document, "cached_feed_entries", None)
+        if cached is not None and len(cached) > 0:
+            document = obj.unified_document
+            content_type_cache = self.context.get('content_type_cache', {})
```

Comment:
This code involves business logic that belongs elsewhere. I suggest we just get rid of caching for now or move it to the view in a clean way.

The serializer should be pretty dumb and not rely on context. Relying on context makes it not reusable and could lead to unexpected behavior. For example, using the serializer without context will result in `content_type_cache` to always be `None` which will default to `target_content_type` of `post`